pkg/controller/load_configuration.go

```diff
@@ -0,0 +1,120 @@
+package controller
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	errs "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// LoadFromSecret retrieves an operator secret and sets environment
+// variables in order to override default configurations.
+// If no secret is found, then configuration will use defaults.
+// Returns error if WATCH_NAMESPACE is not set, if the resource GET request failed
+// (for other reasons apart from isNotFound) and if setting env vars fails.
+//
+// prefix: represents the operator prefix (HOST_OPERATOR/MEMBER_OPERATOR)
+// resourceKey: is the env var which contains the secret resource name.
+// cl: is the client that should be used to retrieve the configmap.
+func LoadFromSecret(prefix, resourceKey string, cl client.Client) error {
+	// get the secret name
+	secretName := getResourceName(resourceKey)
+	if secretName == "" {
+		return nil
+	}
+
+	// get the secret
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		return err
+	}
+
+	secret := &v1.Secret{}
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: secretName}
+	err = cl.Get(context.TODO(), namespacedName, secret)
+	if err != nil {
+		if !errs.IsNotFound(err) {
+			return err
+		}
+		logf.Log.Info("secret is not found")
+	}
+
+	// get secrets and set environment variables
+	for key, value := range secret.Data {
+		secretKey := createOperatorEnvVarKey(prefix, key)
+		err := os.Setenv(secretKey, string(value))
```

Comment:
@tinakurian 
We need to be sure that none of the operators prints the secret variables to the log.
host-operator seems to be OKish: https://github.com/codeready-toolchain/host-operator/blob/master/cmd/manager/main.go#L277-L289
but member-operator is not safe: https://github.com/codeready-toolchain/member-operator/blob/master/cmd/manager/main.go#L254-L260 this has to be fixed, otherwise, the secrets will be printed to the log.

Btw, do we really need to set them as env variables? Why don't you store them as variables of the configuration object? This shouldn't be so hard (even if you want to make it generic) and it will be much safer. I have some weird feeling when we set the secrets as normal variables.

How I would do it? There are  a few options, but the easiest way is to store them in a map:
```go
type Config struct {
	host  *viper.Viper
	secretValues map[string]string
}
```
by default, the map would already contain the keys and the default values.
then then `LoadFromSecret` would take the map and assign the values there (if not defined then it keeps the default value). 
The respective methods (such as `GetMailgunDomain()`) would then read the value from the map

Reply:
Good points @MatousJobanek .

Reply:
@MatousJobanek i don't think member-operator has secrets at the moment? just a configmap

Reply:
@MatousJobanek i have changed this to use a map as you suggested, please check this PR: https://github.com/codeready-toolchain/host-operator/pull/255. I've also updated this PR to reflect those changes: deab6ebe13dfe4de125b50d196c3b414be22630a

Reply:
Awesome, thanks a lot for this :+1: 
So, from the code, it looks like you delegate the "default values management" to the `Get*` methods, right?

Reply:
@MatousJobanek no worries! currently these values don't have any defaults. they either exist via the secret or they don't exist at all. but in the future, if we needed a secret value to have a default it would need to be handled in the configuration `Get*` funcs